src/WordPress/Hooks.php

```diff
@@ -154,7 +155,12 @@ public function purgeCacheByRelevantURLs($postIds)
             $urls = [];
             foreach ($postIds as $postId) {
                 // Do not purge for autosaves or updates to post revisions.
-                if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
+                if (wp_is_post_autosave($postId) || wp_is_post_revision($postId) || (defined('DOING_AUTOSAVE') && DOING_AUTOSAVE)) {
+                    continue;
+                }
+
+                // Do no purge for non-published posts (auto-draft, draft)
+                if (get_post_status($postId) !== 'publish') {
```

Comment:
this is also in #474. can you please clean them up to ensure the PR is only targeting the change in question? we don't want to be resolving merge conflicts here.

Reply:
i'd prefer small, specific PRs for single changes over big PRs. easier review makes it more straight forward for maintainers to understand the change.